include/alpaka/mem/buf/Traits.hpp

```diff
@@ -191,4 +191,82 @@ namespace alpaka
 
         ALPAKA_UNREACHABLE(allocBuf<TElem, TIdx>(host, extent));
     }
+
+    namespace detail
+    {
+        template<typename DevDst, typename DevSrc>
+        auto canZeroCopy(const DevDst& devDst, const DevSrc& devSrc) -> bool
```

Comment:
can this be `constexpr` ?

Reply:
Maybe. I chose to be conservative for now in case we ever need to call a runtime function to figure out whether we can zero-copy between two devices.

Reply:
Understood.

By the way, what should be the behaviour of `zero_memcpy(queue, bufOnSomeGpu, bufOnOtherGPu);` ?

Make a copy, or let `SomeGpu` access the buffer from `OtherGpu` memory through nvlink/PCIe/etc. ?

Reply:
That is a very good question and I don't know what's best. I added it to #28.

Reply:
I would prefer the second version as it makes use of (fast) hardware capabilities. Now the question to me is: Can and should we decide if this is possible automatically or let the users declare what they want

Reply:
So, I renamed the feature `makeAvailable` now, which is what its intend is. So `makeAvailable(queue, bufOnSomeGpu, bufOnOtherGPu)` should make the memory available on another GPU. If zero-copying is possible by any chance (mapping memory through PCIe, nvlink etc), it should be done. If non of that is possible, a full copy is performed.

If the user intends to have a full copy anyway, they should just say so and write `memcpy(queue, bufOnSomeGpu, bufOnOtherGPu)`. In that sense, we should decide this automatically. But that decision may depend on the user having declared the involved buffers in a certain way.

If the user intended to use a buffer over nvlink between 2 CUDA CPUs (a very explicit use), then they don't need to use alpaka and can just call the CUDA API directly.